scripts/bert/finetune_classifier.py

```diff
@@ -46,41 +47,93 @@
 from gluonnlp.model import bert_12_768_12
 from gluonnlp.data import BERTTokenizer
 
-from bert import BERTClassifier
-from dataset import MRPCDataset, BERTDatasetTransform
-
-parser = argparse.ArgumentParser(description='BERT sentence pair classification example.'
-                                             'We fine-tune the BERT model on MRPC')
-parser.add_argument('--epochs', type=int, default=3, help='number of epochs, default is 3')
-parser.add_argument('--batch_size', type=int, default=32,
-                    help='Batch size. Number of examples per gpu in a minibatch, default is 32')
-parser.add_argument('--dev_batch_size', type=int, default=8,
-                    help='Batch size for dev set, default is 8')
-parser.add_argument('--optimizer', type=str, default='adam',
-                    help='Optimization algorithm, default is adam')
-parser.add_argument('--lr', type=float, default=5e-5,
-                    help='Initial learning rate, default is 5e-5')
-parser.add_argument('--warmup_ratio', type=float, default=0.1,
-                    help='ratio of warmup steps used in NOAM\'s stepsize schedule, default is 0.1')
-parser.add_argument('--log_interval', type=int, default=10, help='report interval, default is 10')
-parser.add_argument('--max_len', type=int, default=128,
-                    help='Maximum length of the sentence pairs, default is 128')
-parser.add_argument('--seed', type=int, default=2, help='Random seed, default is 2')
-parser.add_argument('--accumulate', type=int, default=None, help='The number of batches for '
-                    'gradients accumulation to simulate large batch size. Default is None')
-parser.add_argument('--gpu', action='store_true', help='whether to use gpu for finetuning')
+from bert import BERTClassifier, BERTRegression
+from dataset import MRPCDataset, QQPDataset, RTEDataset, \
+    STSBDataset, BERTDatasetTransform, \
+    QNLIDataset, COLADataset, MNLIDataset, WNLIDataset, SSTDataset
+
+tasks = {
+    'MRPC': MRPCDataset,
+    'QQP': QQPDataset,
+    'QNLI': QNLIDataset,
+    'RTE': RTEDataset,
+    'STS-B': STSBDataset,
+    'CoLA': COLADataset,
+    'MNLI': MNLIDataset,
+    'WNLI': WNLIDataset,
+    'SST': SSTDataset
+}
+
+parser = argparse.ArgumentParser(
+    description='BERT fine-tune examples for GLUE tasks.')
+parser.add_argument(
+    '--epochs', type=int, default=3, help='number of epochs, default is 3')
+parser.add_argument(
+    '--batch_size',
+    type=int,
+    default=32,
+    help='Batch size. Number of examples per gpu in a minibatch, default is 32'
+)
+parser.add_argument(
+    '--dev_batch_size',
+    type=int,
+    default=8,
+    help='Batch size for dev set, default is 8')
+parser.add_argument(
+    '--optimizer',
+    type=str,
+    default='adam',
```

Comment:
Shall we change the default optim to bertadam?

Reply:
Yes!